### PR TITLE
DXCDT-60 Add attack protection feature

### PR DESCRIPTION
**Pull request by sergiughf**
### Description

This PR adds functionality to manage the attack protection resource thorough the cli. The flow can be seen in the below gif:

ezgif com-gif-maker (1)

### Testing

- [x] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [x] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not `master`

### CODE REVIEW

**Comment by sergiughf** [comment]
We pair reviewed this with @willvedd and I'm incorporating some feedback we had in our zoom call at the moment. Stay tuned!

File: commander.yaml [hunk 1]
```diff
@@ -571,3 +571,71 @@ tests:
...
+    stdout:
+      contains:
+        - ENABLED
+        - SHIELDS
+        - ALLOW_LIST
+        - STAGE_PRE_LOGIN_MAX_ATTEMPTS
+        - STAGE_PRE_LOGIN_RATE
+        - STAGE_PRE_USER_REGISTRATION_MAX_ATTEMPTS
+        - STAGE_PRE_USER_REGISTRATION_RATE
+    exit-code: 0
```
Comment (willvedd) [inline commander.yaml:641]: Great, I'm glad you were able to make these tests a bit tighter.

File: docs/Gemfile.lock [hunk 2]
```diff
@@ -272,20 +272,21 @@ GEM
...
+  github-pages (~> 225)
   jekyll-feed (~> 0.12)
   minima (~> 2.5)
   tzinfo (~> 1.2)
   tzinfo-data
   webrick (~> 1.7)
 
 BUNDLED WITH
-   2.2.28
+   2.3.3
```
Comment (willvedd) [inline docs/Gemfile.lock:292]: I'm ok with these changes being shimmed into this PR, but given that it's become quite large, I suggest we separate when possible for simpler reviews in the future.

**Review by willvedd — APPROVED** [review]
Great work. Thanks for reviewing with me earlier.

**Comment by cocojoe** [comment]
@sergiughf great seeing videos/gifs in PRs to demonstrate the DX

**Comment by cocojoe** [comment]
@sergiughf in the future can we break out PRs to focus on their goals, I see linting type changes on a bunch of files and shuffling around imports to group them, this is housekeeping and should be a separate PR it just adds noise to the goal of this PR.

Same for dependency bumps, it's "okay" but not the goal of this PR.

**Pull request closed without merging**